### PR TITLE
Feat/extra info

### PR DESCRIPTION
**Pull request by MarkVlamPersie**
- [x] Code review
- [x] Comments
- [x] Functional tests written
- [x] Functional tests checked by someone other than the author(s)
- [x] No linting errors
- [x] Technical documentation written
- [x] Technical documentation reviewed
- [x] Removed unnecessary includes

### CODE REVIEW

File: src/logic/entity/static/building/building_entity.h [hunk 1]
```diff
@@ -20,9 +20,7 @@ class BuildingEntity : public StaticEntity {
     BuildingType building_type;
     float delta_time;
     float order_time;
-    vec2 spawn;
```
Comment (sschrijver) [inline src/logic/entity/static/building/building_entity.h]: Private variables has to start with underscore

File: src/logic/entity/static/building/spawnable_entity.h [hunk 2]
```diff
@@ -5,23 +5,34 @@
...
+#include <string>
+
+class Resources;
+
 class SpawnableEntity {
 private:
     Resources* _cost;
-    MovingEntityType _entity_type;
+
+    MovingEntityType entity_type;
```
Comment (sschrijver) [inline src/logic/entity/static/building/spawnable_entity.h]: underscores

File: src/logic/entity/static/building/spawnable_entity.h [hunk 3]
```diff
@@ -5,23 +5,34 @@
...
+    std::string description;
 public:
     SpawnableEntity(Resources* cost, MovingEntityType entityType);
 
+    SpawnableEntity(Resources* cost, MovingEntityType entityType, std::string description);
```
Comment (sschrijver) [inline src/logic/entity/static/building/spawnable_entity.h]: entity_type

File: src/logic/entity/static/building/warehouse_entity.cpp [hunk 4]
```diff
@@ -2,6 +2,7 @@
 // Created by Sander on 4-4-2017.
 //
 
+#include <iostream>
```
Comment (sschrijver) [inline src/logic/entity/static/building/warehouse_entity.cpp]: unnecessary include?

File: src/renderer/sdl/badge/sdl_badge_render_object.h [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+// Created by Jeroen on 5/24/2017.
+//
+
+#ifndef CONSTRUCT_AND_DESTROY_SDL_BADGE_RENDER_OBJECT_H
+#define CONSTRUCT_AND_DESTROY_SDL_BADGE_RENDER_OBJECT_H
+
+#include <string>
+#include "sdl/sdl_render_object.h"
+#include "types.h"
+#include "sdl/sdl_renderer.h"
```
Comment (sschrijver) [inline src/renderer/sdl/badge/sdl_badge_render_object.h]: Shouldn't SDLRenderer be a forward decleration?

File: src/renderer/sdl/label/sdl_cost_label.cpp [hunk 6]
```diff
@@ -0,0 +1,54 @@
+//
+// Created by Mark on 23-5-2017.
+//
+
+#include <iostream>
```
Comment (sschrijver) [inline src/renderer/sdl/label/sdl_cost_label.cpp]: iostream still needed?

File: src/renderer/sdl/label/sdl_render_label.cpp [hunk 7]
```diff
@@ -2,6 +2,7 @@
 // Created by bouwm on 8-5-2017.
 //
 
+#include <iostream>
```
Comment (sschrijver) [inline src/renderer/sdl/label/sdl_render_label.cpp]: still needed?

File: src/ui/sdl/event/slot/mouse_handler_buildingpanel.cpp [hunk 8]
```diff
@@ -2,6 +2,9 @@
 // Created by Jeroen on 5/11/2017.
 //
 
+#include <iostream>
```
Comment (sschrijver) [inline src/ui/sdl/event/slot/mouse_handler_buildingpanel.cpp]: same here

File: src/ui/sdl/event/slot/mouse_handler_entitypanel.cpp [hunk 9]
```diff
@@ -2,18 +2,18 @@
 // Created by Jeroen on 5/18/2017.
 //
 
+#include <iostream>
```
Comment (sschrijver) [inline src/ui/sdl/event/slot/mouse_handler_entitypanel.cpp]: I don't see any std::cout in this file

File: src/ui/sdl/panel/sdl_control_panel.cpp [hunk 10]
```diff
@@ -2,6 +2,7 @@
 // Created by Jeroen on 5/19/2017.
 //
 
+#include <sdl/label/sdl_name_label.h>
```
Comment (sschrijver) [inline src/ui/sdl/panel/sdl_control_panel.cpp]: "..."

File: src/renderer/sdl/badge/sdl_badge_render_object.cpp [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+void SDL_BadgeRenderObject::update_count(int count) {
+    this->_count = count;
+}
+
+
+void SDL_BadgeRenderObject::update_color(sdl_data *color) {
+    this->_color = color;
+}
```
Comment (SanderBouwman) [inline src/renderer/sdl/badge/sdl_badge_render_object.cpp]: destructor missing

File: src/renderer/sdl/label/sdl_cost_label.cpp [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+        temp->red = 255;
+    } else {
+        temp->blue = data->blue;
+        temp->green = data->green;
+        temp->red = data->red;
+    }
+
+    _image->render(renderer);
+    _text->render(renderer);
+}
```
Comment (SanderBouwman) [inline src/renderer/sdl/label/sdl_cost_label.cpp]: Destructor missing

File: src/renderer/sdl/label/sdl_name_label.cpp [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+
+    sdl_text *temp = (sdl_text*)_text_object->get_data();
+    temp->text = _text.c_str();
+
+    _text_object->render(renderer);
+}
+
+void SDLNameLabel::set_text(std::string text) {
+    this->_text = text;
+}
```
Comment (SanderBouwman) [inline src/renderer/sdl/label/sdl_name_label.cpp]: Destructor missing

File: src/ui/sdl/panel/sdl_entity_panel.h [hunk 14]
```diff
@@ -5,14 +5,24 @@
...
+class SDL_BadgeRenderObject;
+class SDL_ProgressbarRenderObject;
 
 /**
  * Renders panels for all entities that can be spawned
  */
 class SDLEntityPanel : public SDLPanel {
+private:
+    std::map<MovingEntityType, SDL_BadgeRenderObject*> _badges;
+    std::map<MovingEntityType, SDL_ProgressbarRenderObject*> _progressbar_renderers;
```
Comment (SanderBouwman) [inline src/ui/sdl/panel/sdl_entity_panel.h]: _progressbar_renderers isn't used.

File: src/renderer/sdl/badge/sdl_badge_render_object.h [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+class SDL_UI_RenderTextObject;
+
+class SDL_BadgeRenderObject : public SDL_RenderObject{
+private:
+    int _count;
+    int _radius;
+    vec2 _center;
+    sdl_text *_textdata;
+    SDL_UI_RenderTextObject *_text;
+    sdl_data *_color;
```
Comment (SanderBouwman) [inline src/renderer/sdl/badge/sdl_badge_render_object.h]: _color isn't used.

File: src/renderer/sdl/label/sdl_cost_label.cpp (hunk 12 above)
Comment (MarkVlamPersie) [inline src/renderer/sdl/label/sdl_cost_label.cpp]: Made one

File: src/renderer/sdl/label/sdl_name_label.cpp (hunk 13 above)
Comment (MarkVlamPersie) [inline src/renderer/sdl/label/sdl_name_label.cpp]: Made one

**Comment by SanderBouwman** [comment]
Resolve mergeconflicts please.

**Comment by JeroenVinke** [comment]
Done

**Comment by JeroenVinke** [comment]
@MarkVlamPersie when selecting a building, I sometimes get a segfault. Also noticed that at one point the castle showed that it cost 0 stone, which should have been 5

image

can you take a look?

**Comment by sschrijver** [comment]
Commenting this line works for me, but I don't know if this is a good fix:
image

(Line 57 of 'sdl_cost_label.cpp')

**Comment by SanderBouwman** [comment]
delete this->_resource;
should be:
this_resource =nullptr;

**Comment by MarkVlamPersie** [comment]
Segfault has been fixed. The other bug @JeroenVinke mentioned cant be reproduced by me. So it seems fixed by fixing the seg fault.

**Pull request merged**